processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -232,6 +232,9 @@ else if ( !method.isUpdateMethod() ) {
             // determine accessors
             Map<String, Accessor> accessors = resultTypeToMap.getPropertyWriteAccessors( cms );
             this.targetProperties = new LinkedHashSet<>( accessors.keySet() );
+            if ( targetProperties.isEmpty() ) {
```

Comment:
I missed this in my previous review. Looking at this if the target does not have setters but has constructor arguments then reporting it here before the constructor has been processed is incorrect. Shouldn't we move this after `this.targetProperties` is no longer updated (e.g. on line 257 it is populated)? See the adjusted tests.

Reply:
You are right! 
I think we also have to come up with a solution for the newly added warning, which leads to a lot of existing Tests failing.

Reply:
Let me see what kind of tests are failing. Mapping into an empty object is anyways a bit strange. Perhaps we need to adjust those test cases.